store/sqlstore/upgrade.go

```diff
@@ -1211,3 +1212,10 @@ func upgradeDatabaseToVersion537(sqlStore *SqlStore) {
 	// saveSchemaVersion(sqlStore, Version5370)
 	// }
 }
+
+func upgradeDatabaseToVersion600(sqlStore *SqlStore) {
+	sqlStore.GetMaster().ExecNoTimeout("UPDATE Posts SET RootId = ParentId WHERE RootId = '' AND RootId != ParentId")
+	sqlStore.RemoveColumnIfExists("Posts", "ParentId")
+	sqlStore.GetMaster().ExecNoTimeout("UPDATE CommandWebhooks SET RootId = ParentId WHERE RootId = '' AND RootId != ParentId")
+	sqlStore.RemoveColumnIfExists("CommandWebhooks", "ParentId")
```

Comment:
Honestly I don't know if the updates are really needed but added them to be on the safe side.

Reply:
could we cross-check if these updates are really needed.

My only worry here is mostly for MySQL. MySQL (as far I remember for 5.6 and above) locks the table for alter. There is an attribute `LOCK=NONE` for `InnoDB` which allows read and writes during the operation I think.

Could we give a try to see what happens and see if the read/write still works?

Reply:
It's a bit difficult to say whether the updates are really needed or not because it really depends on the customer's DB state. But more importantly, I think the update should not take more than a few seconds since we are searching by a single index. We can easily confirm that. Given the benefit it gives (a cleaner DB state), that seems like a small cost to pay.

> There is an attribute LOCK=NONE for InnoDB which allows read and writes during the operation I think.

That is only for `ALTER TABLE` statements, not `UPDATE TABLE`. And even for those scenarios, it won't work for our case because we use more than two full-text indexes in the Posts table.

Reply:
Those were only added to prevent data loss but I agree that most probably they are not needed. I've checked our community db and we actually have cases (I found 55) where both `RootId` and `ParentId` are set but with different, non-empty values so that case wouldn't be covered by that. If I recall we were experimenting with subthreading at some point so that may be related.

Of course I've checked the update query on a big dataset (25M posts) and it took a couple of minutes which is nothing compared to the main schema change so eventually decided to leave it.